system/security/cryptohelper/pke.cpp

```diff
@@ -73,14 +73,27 @@ void CLoadedKey::finalize(RSA *_rsa, const char *_keyName)
     keyName.set(_keyName);
 }
 
+//called during PEM_read_bio_RSA_PUBKEY to set passphrase
+int passphraseCB(char *passPhraseBuf, int passPhraseBufSize, int rwflag, void *pPassPhraseMB)
```

Comment:
It seems openssl passes in "1024" for the passPhraseBufSize parameter. So a check for the limit should be done in constructor CLoadedPrivateKeyFromMemory(const char *key, size32_t passPhraseLen, const void *passPhrase), so that user would know if they pass in a passphrase longer than that. Also here an exception/error should be thrown or logged just in case...

Reply:
I don't believe there is a hard limit of 1024 bytes here.  OpenSSL does use PEM_BUFFSIZE in most (all?) places where the callback is invoked, and PEM_BUFFSIZE does currently = 1024, but I don't think we should encode that limit on our end.  OpenSSL does have a habit of changing their minds....  That said, it may be a good idea to perform a sanity check on passPhraseBufSize and make sure it's not some unrealistic number.

Reply:
@mayx @dcamper As I stated in the commentary for the callback, we should NOT impose a hardcoded limit, and should instead rely on the mazsize provided by OpenSSL as an input argument to the callback. FWIW, it is unlikely that a passphrase would be that long

Reply:
It's true that in almost all circumstances, the user won't pass in anything longer than PEM_BUFFSIZE. But if they do, either purposely or inadvertently, should the user be told that the password is over openssl's limit and so didn't get set, or just silently not setting it?

Reply:
To be clear, I'm not talking about limiting passPhraseBufsize that is passed in by openssl, I'm talking about the size of the password passed in by the user pPassPhraseMB, if its size is over passPhraseBufsize, should we silently discard the passphrase? And, would it be better to detect earlier on that the password is too long?

Reply:
I agree we should add some message to alert the caller that the passphrase is too long once we detect that, but I won't know that until the callback.  I can throw from in there with a meaningful message

Reply:
We already know openssl limits the buffer to PEM_BUFFSIZE, so we can use it to limit the passphrase. But we probably don't want to expose the openssl specific constant, which can be solved by adding a method getPasswordSizeLimit() maybe?

Reply:
I think it a bad practice to impose a hard limit in our code based one examination of a compile time constant found in a 3rd parties source code.  The reason they return the buffersize in the call back is so we can do the checking there, as is mentioned in the documentation (return 0 if buffer too small). Again, it is unlikely this limit will be hit

Reply:
Throwing or logging a meaningful message in the callback function is probably good enough for all practical purposes.